src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,1065 @@
+/*
+  This file is part of the STM32duinoBLE library.
+  Copyright (c) 2019 STMicroelectronics. All rights reserved.
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#include "HCISharedMemTransport.h"
+#include "STM32Cube_FW/hw.h"
+#include "stm32wbxx_hal_rtc.h"
+#include "stm32wbxx_hal_rtc_ex.h"
+
+/* global var used as semaphore to control incoming events */
+volatile bool sys_event; /* true : M0 core is already up and running */
+volatile bool sys_resp;
+volatile bool acl_data; /* true : data acknowledged by the BLE */
+volatile bool data_available;
```

Comment:
**Any use** of those volatile data needs to be protected against concurrent access with critical sections (enable / disable irqs)

Reply:
Is this part resolved ?

Reply:
removing unused volatile bool sys_resp 
setting sys_event and acl_data_on and data_overflow
is now protected by  __disable_irq(); /    __enable_irq();